### PR TITLE
[vcpkg.cmake] Add default triplet for mingw compilers (world rebuild?)

### PR DESCRIPTION
**Pull request by longnguyen2004**
**Describe the pull request**

- #### What does your PR fix?  
  Automatically set the triplet to `*-mingw-dynamic` for mingw compilers.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  mingw, not needed

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

### CODE REVIEW

**Comment by longnguyen2004** [comment]
How is this PR going?

**Comment by PhoebeHui** [comment]
@longnguyen2004, sorry for the delay, I will ask other team members to help review.

**Comment by PhoebeHui** [comment]
I'm not sure if we would accpet this officially, need other team member take a further look.

**Comment by BillyONeal** [comment]
I want to get other maintainers' feedback before merging this. Should have an answer Thursday.

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -339,6 +339,8 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin" OR (NOT CMAKE_SYSTEM_NAME AND CMAKE_H
     set(Z_VCPKG_TARGET_TRIPLET_PLAT osx)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "iOS")
     set(Z_VCPKG_TARGET_TRIPLET_PLAT ios)
+elseif(MINGW)
+    set(Z_VCPKG_TARGET_TRIPLET_PLAT mingw-dynamic)
```
Comment (BillyONeal) [inline scripts/buildsystems/vcpkg.cmake:343]: Would this be a better experience if we defaulted to `mingw-static` ? /cc @dg0yt

Can we get anyone else who commonly uses MinGW to comment on this?

(After resolving the dynamic default vs. static default question we will merge this)

**Comment by BillyONeal** [comment]
> Ping @longnguyen2004 for your response..

https://github.com/microsoft/vcpkg/pull/19020#discussion_r724950914 is the response. I would like other MinGW folks to comment but otherwise this seems reasonable.

**Comment by PhoebeHui** [comment]
cc @dg0yt @autoantwort @talregev  @fairlyblank  @past-due @Pospelove

**Comment by Pospelove** [comment]
LGTM

**Pull request merged**